aws/resource_aws_vpc_peering_connection.go

```diff
@@ -238,33 +219,32 @@ func resourceAwsVPCPeeringUpdate(d *schema.ResourceData, meta interface{}) error
 		return nil
 	}
 
-	pc := pcRaw.(*ec2.VpcPeeringConnection)
-
-	if _, ok := d.GetOk("auto_accept"); ok {
-		if pc.Status != nil && *pc.Status.Code == ec2.VpcPeeringConnectionStateReasonCodePendingAcceptance {
-			status, err := resourceVPCPeeringConnectionAccept(conn, d.Id())
-			if err != nil {
-				return fmt.Errorf("Unable to accept VPC Peering Connection: %s", err)
-			}
-			log.Printf("[DEBUG] VPC Peering Connection accept status: %s", status)
+	if _, ok := d.GetOk("auto_accept"); ok && statusCode == ec2.VpcPeeringConnectionStateReasonCodePendingAcceptance {
+		statusCode, err = resourceVPCPeeringConnectionAccept(conn, d.Id())
+		if err != nil {
+			return fmt.Errorf("Unable to accept VPC Peering Connection: %s", err)
 		}
+		log.Printf("[DEBUG] VPC Peering Connection accept status: %s", statusCode)
 	}
 
 	if d.HasChange("accepter") || d.HasChange("requester") {
-		_, ok := d.GetOk("auto_accept")
-		if !ok && pc.Status != nil && *pc.Status.Code != "active" {
+		if statusCode == ec2.VpcPeeringConnectionStateReasonCodeActive || statusCode == ec2.VpcPeeringConnectionStateReasonCodeProvisioning {
```

Comment:
VPC peering connection is in `provisioning` state if it has just been created.
DNS options can be set if the connection is in this state.